files/src/main/scala/io/iteratee/files/EffectCapture.scala

```diff
@@ -0,0 +1,35 @@
+package io.iteratee.files
+
+import cats.ApplicativeError
+import cats.data.Xor
+import scala.util.Try
+import scala.util.control.NonFatal
+
+trait EffectCapture[F[_]] {
+  def apply[A](a: => A): F[A]
```

Comment:
can we comment this at all? Seems awfully vague. Is `a` evaluated each time there is a `.map` or `flatMap` of `F[A]`? If so, this disqualifies `Xor` and `Try`. If not, why isn't this just `catchNonFatal` in practice, which different `MonadError[F, Throwable]` can override as they see fit?